timemachine/fe/aligned_potential.py

```diff
@@ -0,0 +1,398 @@
+
+    # parameters for pairs that do not interact in the dst state
+    w = interpolate_w_coord(src_w, cutoff, lamb)
+    pair_params_excluded_dst = jnp.concatenate((src_qlj, w[:, None]), axis=1)
+
+    # parameters for pairs that interact in both src and dst states
+    w = jax.vmap(interpolate.linear_interpolation, (0, 0, None))(src_w, dst_w, lamb)
+    qlj = interpolate.linear_interpolation(src_qlj, dst_qlj, lamb)
+    pair_params_not_excluded = jnp.concatenate((qlj, w[:, None]), axis=1)
+
+    pair_params = jnp.where(
+        is_excluded_src,
+        pair_params_excluded_src,
+        jnp.where(
+            is_excluded_dst,
+            pair_params_excluded_dst,
+            pair_params_not_excluded,
+        ),
+    )
+
+    return pair_params
+
+
+@dataclass(kw_only=True)
+class AlignedPotential:
+    src_params: Union[NDArray[np.float64], jax.Array]
+    dst_params: Union[NDArray[np.float64], jax.Array]
+    mins: Optional[NDArray[np.float64]]
+    maxes: Optional[NDArray[np.float64]]
+
+    def interpolate(self, lamb):
+        raise NotImplementedError()
+
+
+@dataclass(kw_only=True)
+class AlignedBond(AlignedPotential):
+    idxs: NDArray[np.int32]
+
+    def interpolate(self, lamb):
+        k_min = 0.1
+        params = batch_interpolate_harmonic_bond_params(
+            self.src_params, self.dst_params, lamb, k_min, self.mins, self.maxes
+        )
+        params = jnp.array(params).T
+
+        return HarmonicBond(self.idxs).bind(params)
+
+
+@dataclass(kw_only=True)
+class AlignedAngle(AlignedPotential):
+    idxs: NDArray[np.int32]
+
+    def interpolate(self, lamb):
+        k_min = 0.05
+        params = batch_interpolate_harmonic_angle_params(
+            self.src_params, self.dst_params, lamb, k_min, self.mins, self.maxes
+        )
+        params = jnp.array(params).T
+        return HarmonicAngle(self.idxs).bind(params)
+
+
+@dataclass(kw_only=True)
+class AlignedTorsion(AlignedPotential):
+    idxs: NDArray[np.int32]
+
+    def interpolate(self, lamb):
+        params = batch_interpolate_periodic_torsion_params(
+            self.src_params, self.dst_params, lamb, self.mins, self.maxes
+        )
+        params = jnp.array(params).T
+        return PeriodicTorsion(self.idxs).bind(params)
+
+
+@dataclass(kw_only=True)
+class AlignedChiralAtom(AlignedPotential):
+    idxs: NDArray[np.int32]
+
+    def interpolate(self, lamb):
+        k_min = 0.025
+        params = batch_interpolate_chiral_atom_params(
+            self.src_params, self.dst_params, lamb, k_min, self.mins, self.maxes
+        )
+        params = jnp.array(params).reshape(-1)
+        return ChiralAtomRestraint(self.idxs).bind(params)
+
+
+@dataclass(kw_only=True)
+class AlignedNonbondedPairlist(AlignedPotential):
+    idxs: NDArray[np.int32]
+    cutoff: float
+    beta: float
+
+    def interpolate(self, lamb):
+        # (ytz): batch_interpolate_nonbonded_pair_list_params currently fails to respect the self.mins and self.maxes
```

Comment:
Worth asserting that `(self.mins is None and self.maxes is None) or np.all((self.mins == 0.0) & (self.maxes == 1.0))` to avoid surprises if this is unsupported?